### PR TITLE
[Cleanup] Quantum Painter

### PR DESCRIPTION
**Pull request by elpekenin**
## Description

Just some tidying of QP's codebase, adding some `typedef`s here and there to get rid of boilerplate `struct` Some of these types are defined in `.c` files, i dont know if they should be moved to `.h`, i also renamed a couple of them, appending `_t`

A couple of minor changes too
  - Replace a ternary with `QP_MIN`
  - Replace ` __attribute__((packed))` with `QP_PACKED`

Tested by compiling my keyboard's files and display worked fine. Didn't test exhaustively tho

~~Haven't linted, as i think the changes haven't mess up formatting, im gonna let CI warn and fix offending files if any.~~

## Types of Changes

- [X] Core
- [ ] Bugfix
- [ ] New feature
- [X] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]
From memory, everything `struct` was QP-internal.
Guess it doesn't really matter.

**Comment by tzarc** [comment]
SHA1's all match for all QP-enabled boards in the repo.

**Comment by elpekenin** [comment]
> SHA1's all match for all QP-enabled boards in the repo.

What does this mean? "Everything compiles"? 😅

**Comment by tzarc** [comment]
> > SHA1's all match for all QP-enabled boards in the repo.
> 
> What does this mean? "Everything compiles"? 😅

Yes, generated binaries are identical before and after this PR is applied.

**Comment by drashna** [comment]
Merge conflicts here.

File: quantum/painter/qp_internal_driver.h [hunk 1]
```diff
@@ -81,9 +81,4 @@ struct painter_driver_t {
 
     // Comms config pointer -- needs to point to an appropriate comms config if the comms driver requires it.
     void *comms_config;
-};
-
-////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-// Device internals
-
-bool qp_internal_register_device(painter_device_t driver);
+} painter_driver_t;
```
Comment (tzarc) [inline quantum/painter/qp_internal_driver.h:84]:
```suggestion
} painter_driver_t;

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// Device internals

bool qp_internal_register_device(painter_device_t driver);
```

**Review by tzarc — CHANGES_REQUESTED** [review]

**Review by tzarc — APPROVED** [review]

**Comment by tzarc** [comment]
SHA1's still match.

**Review by KarlK90 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**